dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java

```diff
@@ -1406,26 +1412,40 @@ private String getEntityTypeStringFromMetadata(List<MetadataValue> list) {
         return entityType;
     }
 
-    private MetadataValue constructResultingMetadataValue(Context context, Item item, Item otherItem, VirtualBean virtualBean,
-                                                          MetadataValue metadataValue) throws SQLException {
+    private RelationshipMetadataValue constructResultingMetadataValue(Context context, Item item, Item otherItem,
+                                                                      VirtualBean virtualBean,
+                                                                      RelationshipMetadataValue metadataValue,
+                                                                      Integer relationshipId)
+        throws SQLException {
         metadataValue.setValue(virtualBean.getValue(context, otherItem));
-        metadataValue.setAuthority("virtual");
+        metadataValue.setAuthority("virtual::" + relationshipId);
         metadataValue.setConfidence(-1);
         metadataValue.setDSpaceObject(item);
         return metadataValue;
     }
 
-    private MetadataValue constructMetadataValue(String key) {
-        String[] splittedKey = key.split("\\.");
-        MetadataValue metadataValue = new MetadataValue();
-        MetadataField metadataField = new MetadataField();
-        MetadataSchema metadataSchema = new MetadataSchema();
-        metadataSchema.setName(splittedKey.length > 0 ? splittedKey[0] : null);
-        metadataField.setMetadataSchema(metadataSchema);
-        metadataField.setElement(splittedKey.length > 1 ? splittedKey[1] : null);
-        metadataField.setQualifier(splittedKey.length > 2 ? splittedKey[2] : null);
-        metadataValue.setMetadataField(metadataField);
-        metadataValue.setLanguage(Item.ANY);
-        return metadataValue;
+    private RelationshipMetadataValue constructMetadataValue(Context context, String key) {
+        try {
+            String[] splittedKey = key.split("\\.");
+            RelationshipMetadataValue metadataValue = new RelationshipMetadataValue();
+            String metadataSchema = splittedKey.length > 0 ? splittedKey[0] : null;
+            String metadataElement = splittedKey.length > 1 ? splittedKey[1] : null;
+            String metadataQualifier = splittedKey.length > 2 ? splittedKey[2] : null;
+            MetadataField metadataField = metadataFieldService
+                .findByElement(context, metadataSchema, metadataElement, metadataQualifier);
+            if (metadataField == null) {
+                log.error(
+                    "A MetadataValue was attempted to construct with MetadataField for paremeters: metadataschema: "
+                        + metadataSchema + ", metadataelement:" + metadataElement +
+                        ", metadataqualifier: " + metadataQualifier);
```

Comment:
Since we use log4j v2, we can now format this log statement without string concatenation, like this:
`log.error("A MetadataValue was attempted to construct with MetadataField for parameters: metadataschema: {}, metadataelement: {}, metadataqualifier: {}", metadataSchema, metadataElement, metadataQualifier);

Reply:
This has been updated.
A string concatenation is still required to avoid exceeding the max line length, but this is now a concatenation of 2 regular strings which will be processed on compilation time